Jellyfin.Api/Helpers/AudioHelper.cs

```diff
@@ -154,4 +159,44 @@ public async Task<ActionResult> GetAudioStream(
             transcodingJobType,
             cancellationTokenSource).ConfigureAwait(false);
     }
+
+    /// <summary>
+    /// Gets audio data to draw a waveform bar.
+    /// </summary>
+    /// <param name="itemId">The item id.</param>
+    /// <returns>A <see cref="FileStream"/> containing the resulting .dat file.</returns>
+    public async Task<FileStream> GetAudioWaveForm(Guid itemId)
+    {
+        var item = _libraryManager.GetItemById<BaseItem>(itemId)
+            ?? throw new ResourceNotFoundException();
+
+        var libraryOptions = BaseItem.LibraryManager.GetLibraryOptions(item);
+        var saveInMediaFolder = libraryOptions.SaveLocalMetadata;
+        var saveFileName = Path.GetFileNameWithoutExtension(item.Path) + ".csv";
+        string outputPath = saveInMediaFolder
+            ? Path.Combine(item.ContainingFolderPath, saveFileName)
+            : Path.Combine(item.GetInternalMetadataPath(), saveFileName);
+
+        if (!File.Exists(outputPath))
+        {
+            var processStartInfo = new ProcessStartInfo("ffprobe", $"-v error -select_streams a:0 -show_entries stream=sample_rate -of csv=p=0 \"{item.Path}\"") { RedirectStandardOutput = true, UseShellExecute = false };
+            int sampleRate = 0;
+            using (var process = Process.Start(processStartInfo))
+            {
+                if (process != null)
+                {
+                    string output = await process.StandardOutput.ReadToEndAsync().ConfigureAwait(false);
+                    // This value sets the length of a frame, the amount of measurements per second.
+                    int samplesPerSecond = 2;
+                    sampleRate = Math.Max(int.Parse(output, System.Globalization.CultureInfo.InvariantCulture) / samplesPerSecond, 1);
+                }
+            }
+
+            await Task.Run(() => Process.Start("ffprobe", $"-v error -f lavfi -i \"amovie={item.Path},asetnsamples={sampleRate},astats=metadata=1:reset=1\" -show_entries frame_tags=lavfi.astats.Overall.RMS_peak -of csv=p=0 -o \"{outputPath}\"").WaitForExit()).ConfigureAwait(false);
+        }
+
+        var fileStream = new FileStream(outputPath, FileMode.Open, FileAccess.Read, FileShare.Read);
+
+        return fileStream;
```

Comment:
To make this production ready I really don't think serve the csv as is a good enough. We need some sort of server side process to make the client developer's life easier. For example make it a json response to match the behavior of may other apis, this would also allow us to extend this api without breaking all clients. The JSON should at least the sample duration and the sample array consists all samples, so that the client doesn't have to hard code an early development decision about the duration because that is subject to change. The actual values being measured will be discussed separately because what ffmpeg returns may not be the best to plot a soundwave.